.github/workflows/docker-publish.yml

```diff
@@ -1,96 +1,78 @@
 name: docker
 
 on:
-    push:
-      tags:
-        - "v*.*.*"
-    schedule:
-      - cron: "0 0 * * *"
-    # Trigger without any parameters a proactive rebuild
-    workflow_dispatch: {}
-    workflow_call:
+  push:
+    tags:
+      - "v*.*.*"
+  schedule:
+    - cron: "0 0 * * *"
+  # Trigger without any parameters a proactive rebuild
+  workflow_dispatch: {}
+  workflow_call:
 
 env:
-    REGISTRY: ghcr.io
-    # Will resolve to foundry-rs/foundry
-    IMAGE_NAME: ${{ github.repository }}
+  REGISTRY: ghcr.io
+  # Will resolve to foundry-rs/foundry
+  IMAGE_NAME: ${{ github.repository }}
 
 jobs:
-    container:
-        runs-on: Linux-20.04
-        # https://docs.github.com/en/actions/reference/authentication-in-a-workflow
-        permissions:
-            id-token: write
-            packages: write
-            contents: read
-        timeout-minutes: 120
-        steps:
-            - name: Checkout repository
-              id: checkout
-              uses: actions/checkout@v4
+  build:
+    name: build and push
+    runs-on: ubuntu-20.04
+    permissions:
+      id-token: write
+      packages: write
+      contents: read
+    timeout-minutes: 120
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+      - uses: taiki-e/install-action@cross
```

Comment:
not too familiar with this action but I see there's also `taiki-e/setup-cross-toolchain-action@v1`, mind to explain why this was used instead?

Reply:
Copied from https://github.com/paradigmxyz/reth/blob/343bee568fa8f496a07e8849cb5dad726ba15f41/.github/workflows/docker.yml#L32, I think it used to run `cargo install cross`

Reply:
It uses `binstall` so it downloads binaries instead of compiling them.